src/library/scala/collection/immutable/Range.scala

```diff
@@ -609,15 +609,17 @@ private class RangeIterator(
   }
 
   override def drop(n: Int): Iterator[Int] = {
-    val longPos = _next.toLong + step * n
-    if (step > 0) {
-      _next = Math.min(lastElement, longPos).toInt
-      _hasNext = longPos <= lastElement
-    }
-    else if (step < 0) {
-      _next = Math.max(lastElement, longPos).toInt
-      _hasNext = longPos >= lastElement
+    if (n <= 0) this else {
+      val longPos = _next.toLong + step * n
+      if (step > 0) {
+        _next = Math.min(lastElement, longPos).toInt
+        _hasNext = longPos <= lastElement
+      }
+      else if (step < 0) {
+        _next = Math.max(lastElement, longPos).toInt
+        _hasNext = longPos >= lastElement
+      }
+      this
```

Comment:
It would be more natural to write
```
if (n > 0) {
  sideEffect()
}
this
```

Reply:
I've updated the Range#drop to implement in this form. Thank you for the feedback on this.